### PR TITLE
Non macro thx.fp.List#fromArray

### PR DESCRIPTION
**Pull request by 0b1kn00b**
case Nil        : Cons(next,Nil);
      case Cons(x,xs) : Cons(next,Cons(x,xs));

### CODE REVIEW

File: src/thx/fp/List.hx [hunk 1]
```diff
@@ -16,7 +16,15 @@ abstract List<A>(ListImpl<A>) from ListImpl<A> to ListImpl<A> {
 
   @:from
   inline static public function fromArray<A>(arr : Array<A>) : List<A>
-    return arr.reduceRight.fn(bin(_1, _0), empty());
+    return arr.reduceRight(
+      function(memo,next){
+        return switch(memo){
```
Comment (nuttycom) [inline src/thx/fp/List.hx]: This switch is unnecessary. You could just write `arr.reduceRight(function(memo, next) return Cons(next, memo)`.

Comment (nuttycom) [inline src/thx/fp/List.hx]: Or if you want to get really fancy, `arr.reduceRight(thx.fp.Functions.flip(Cons), empty())`

Comment (nuttycom) [inline src/thx/fp/List.hx]: Oh, except that flip expects a curried function, which Cons is not. So you'd have to define a 'flip2' which has type `((A, B) -> C) -> ((B, A) -> C)`

Comment (0b1kn00b) [inline src/thx/fp/List.hx]: added into #200

**Pull request merged**